### PR TITLE
compact scene hierarchy on load

### PR DESCRIPTION
**Pull request by snoyer**
### Describe your changes

Make the scene tree UI more compact on load by collapsing subtrees that don't contain any meaningful user-provided labels.

### Checklist for finalizing the PR

- [x] I have performed a [self-review](https://f3d.app/dev/CODING_STYLE) of my code
- [x] I have added [tests](https://f3d.app/dev/TESTING) for new features and bugfixes
- [ ] I have added [documentation](https://f3d.app/docs/next/user/QUICKSTART) for new features
- [ ] If it is a modifying the libf3d API, I have updated bindings
- [ ] If it is a modifying the `.github/workflows/versions.json`, I have updated `docker_timestamp`

### Continuous integration

Please write a comment to run CI, eg: `\ci fast`.
See [here](https://f3d.app/dev/CONTRIBUTING#continuous-integration) for more info.

### CODE REVIEW

**Comment by snoyer** [comment]
\ci fast

**Comment by snoyer** [comment]
\ci main

File: vtkext/private/module/vtkF3DMetaImporter.cxx [hunk 1]
```diff
@@ -22,6 +23,98 @@
...
+public:
+  static collapseOnLoadVisitor* New();
+  vtkTypeMacro(collapseOnLoadVisitor, vtkDataAssemblyVisitor);
+
+protected:
+  void SetAttr(int nodeid, bool val)
+  {
+    vtkDataAssembly* mutableAssembly = const_cast<vtkDataAssembly*>(this->GetAssembly());
+    mutableAssembly->SetAttribute(nodeid, "f3d_collapsed", val ? 1 : 0);
+    // TODO is there a way to _remove_ the attr instead of setting it to `0`?
```
Comment (mwestphal) [inline vtkext/private/module/vtkF3DMetaImporter.cxx]: todo ?

File: vtkext/private/module/vtkF3DMetaImporter.cxx [hunk 2]
```diff
@@ -22,6 +23,98 @@
 #include <numeric>
 #include <vector>
 
+namespace
+{
+
+/**
+ * Sets the `f3d_collapsed` attribute on nodes which have
+ * all their children unnamed or named the same as themselves.
```
Comment (mwestphal) [inline vtkext/private/module/vtkF3DMetaImporter.cxx:31]: please doc this default behavior in the user doc, in a concise manner.

File: testing/data/vtk-dasm-test.glb
Comment (mwestphal) [inline testing/data/vtk-dasm-test.glb:1]: I suppose you created this data ?

File: vtkext/private/module/vtkF3DMetaImporter.cxx (hunk 1 above)
Comment (snoyer) [reply vtkext/private/module/vtkF3DMetaImporter.cxx]: I should have written "ask a VTK expert if the Assembly API really has no method to remove an attribute or if I just can't find it" :D

File: testing/data/vtk-dasm-test.glb
Comment (snoyer) [reply testing/data/vtk-dasm-test.glb:1]: correct, I originally sent it on Discord and @Meakk added it as a test, here it's updated with more corner cases

Comment (mwestphal) [reply testing/data/vtk-dasm-test.glb:1]: nice :)

File: vtkext/private/module/vtkF3DMetaImporter.cxx (hunk 1 above)
Comment (Meakk) [reply vtkext/private/module/vtkF3DMetaImporter.cxx]: There is no way to remove an attribute as far as I can tell by reading `vtkDataAssembly.cxx` file

File: vtkext/private/module/vtkF3DMetaImporter.cxx [hunk 3]
```diff
@@ -22,6 +23,88 @@
...
+ * Allows to make the tree more compact on load by collapsing subtrees
+ * that don't contain any meaningful user-provided labels.
+ */
+class collapseOnLoadVisitor : public vtkDataAssemblyVisitor
```
Comment (Meakk) [inline vtkext/private/module/vtkF3DMetaImporter.cxx]: Let's call the class `vtkF3DCollapseOnLoadVisitor`

File: vtkext/private/module/vtkF3DMetaImporter.cxx [hunk 4]
```diff
@@ -22,6 +23,88 @@
...
+    {
+      return std::none_of(childrenIds.cbegin(), childrenIds.cend(),
+        [&](int id) { return this->GetAssembly()->HasAttribute(id, "label"); });
+    };
+
+    const auto allChildrenHaveSameNameAsNode = [&]()
+    {
+      const char* nodeName = this->GetAssembly()->GetAttributeOrDefault(nodeid, "label", "");
+      return std::all_of(childrenIds.cbegin(), childrenIds.cend(), [&](int id)
+        { return !strcmp(this->GetAssembly()->GetAttributeOrDefault(id, "label", ""), nodeName); });
```
Comment (Meakk) [inline vtkext/private/module/vtkF3DMetaImporter.cxx]: Please avoid `strcmp`
```suggestion
      const std::string_view nodeName = this->GetAssembly()->GetAttributeOrDefault(nodeid, "label", "");
      return std::all_of(childrenIds.cbegin(), childrenIds.cend(), [&](int id)
        { return nodeName == this->GetAssembly()->GetAttributeOrDefault(id, "label", ""); });
```

**Comment by mwestphal** [comment]
\ci full

File: vtkext/private/module/vtkF3DMetaImporter.cxx (hunk 2 above)
Comment (mwestphal) [reply vtkext/private/module/vtkF3DMetaImporter.cxx:31]: not adressed yet :)
